rpc/src/client/transport/http.rs

```diff
@@ -32,14 +33,50 @@ use std::io::Read;
 ///     println!("Got ABCI info: {:?}", abci_info);
 /// }
 /// ```
-///
-/// [`Client`]: trait.Client.html
-/// [`Event`]: ./event/struct.Event.html
-/// [`WebSocketClient`]: struct.WebSocketClient.html
 #[derive(Debug, Clone)]
 pub struct HttpClient {
-    host: String,
-    port: u16,
+    inner: sealed::HttpClient,
+}
+
+impl HttpClient {
+    /// Construct a new Tendermint RPC HTTP/S client connecting to the given
+    /// URL.
+    pub fn new<U>(url: U) -> Result<Self>
+    where
+        U: TryInto<HttpClientUrl, Error = Error>,
```

Comment:
Nevermind, I just saw the two calls to `try_into` on lines 51 and 53, meaning that even if the given value is of type `HttpClientUrl`, the conversion into a `hyper::Uri` can still fail.

---

I think using `TryInto` here instead of `Into` or just the actual type is a bit anti-modular.

With `TryInto`, if I already hold a value of type `HttpClientUrl`, I will still be forced to deal with the potential error returned by `new` because of the inner call to `try_into` even if it cannot fail.

While I agree that it's a nice way to keep the calling code short, I think it should be the responsibility of the caller to pass in a valid `HttpClientUrl` (or perhaps a type which implements an infallible `Into<HttpClientUrl>`.

By changing the bound to `Into` (or by just using `HttpClientUrl` since we don't have any `From` instance for this type at the moment), we would go from:

```rust
let literal_case = HttpClient::new("https://localhost:9091")?;

let url: HttpClientUrl = ???;
let known_good_value = HttpClient::new(known_good_value)?;
```

to

```rust
let literal_case = HttpClient::new("https://localhost:9091".parse()?); // moved the error handling one level up

let url: HttpClientUrl = ???;
let known_good_value = HttpClient::new(known_good_value); // no need to deal with error here
```